### PR TITLE
Implementing Guarding the Way

### PR DESCRIPTION
**Pull request by supercodepoet**
image

I didn't see in the tests for a way to check for Sentinel other than just attacking so that is the direction I went.

### CODE REVIEW

File: server/game/cards/03_TWI/events/GuardingTheWay.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    protected override getImplementationId() {
+        return {
+            id: '2359136621',
+            internalName: 'guarding-the-way',
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.setEventAbility({
+            title: 'Give Sentinel for the phase and optionally +2/+2 if you have initiative',
```
Comment (AMMayberry1) [inline server/game/cards/03_TWI/events/GuardingTheWay.ts:15]: nit, wording: "optional" for us means "you can choose not to," which is not the case here

File: test/server/cards/03_TWI/events/GuardingTheWay.spec.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+                context.player1.clickCard(context.guardingTheWay);
+                expect(context.player1).toBeAbleToSelectExactly([context.lukeSkywalkerJediKnight, context.chewbaccaPykesbane]);
+
+                context.player1.clickCard(context.lukeSkywalkerJediKnight);
+                expect(context.lukeSkywalkerJediKnight.getPower()).toBe(8);
+                expect(context.lukeSkywalkerJediKnight.getHp()).toBe(9);
+                expect(context.player2).toBeActivePlayer();
+
+                context.player2.clickCard(context.chewbaccaPykesbane);
```
Comment (AMMayberry1) [inline test/server/cards/03_TWI/events/GuardingTheWay.spec.ts:26]: We should add a comment explaining that this attack is validating the sentinel keyword

File: test/server/cards/03_TWI/events/GuardingTheWay.spec.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+                });
+
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.chewbaccaPykesbane);
+
+                context.player2.clickCard(context.guardingTheWay);
+                expect(context.player2).toBeAbleToSelectExactly([context.lukeSkywalkerJediKnight, context.chewbaccaPykesbane]);
+
+                context.player2.clickCard(context.lukeSkywalkerJediKnight);
```
Comment (AMMayberry1) [inline test/server/cards/03_TWI/events/GuardingTheWay.spec.ts:54]: Also need to do the Sentinel check in this test to confirm that it was applied correctly

File: test/server/cards/03_TWI/events/GuardingTheWay.spec.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+                    },
+                    player2: {
+                        hand: ['guarding-the-way'],
+                        groundArena: ['luke-skywalker#jedi-knight']
+                    }
+                });
+
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.chewbaccaPykesbane);
```
Comment (AMMayberry1) [inline test/server/cards/03_TWI/events/GuardingTheWay.spec.ts:49]: What's the reason for starting Chewbacca in hand?

File: test/server/cards/03_TWI/events/GuardingTheWay.spec.ts [hunk 5]
```diff
@@ -0,0 +1,60 @@
+describe('Guarding the Way', function () {
+    integration(function (contextRef) {
+        describe('Guarding the Way\'s ability', function () {
+            it('should give a unit Sentinel and +2/+2 for the phase', function () {
```
Comment (AMMayberry1) [inline test/server/cards/03_TWI/events/GuardingTheWay.spec.ts:4]: nit, clarity: should include "when the controller has initiative"

File: test/server/cards/03_TWI/events/GuardingTheWay.spec.ts [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+                context.player2.clickCard(context.chewbaccaPykesbane);
+                expect(context.chewbaccaPykesbane.damage).toBe(8);
+                expect(context.lukeSkywalkerJediKnight.damage).toBe(4);
+
+                context.moveToNextActionPhase();
+                expect(context.lukeSkywalkerJediKnight.getPower()).toBe(6);
+                expect(context.lukeSkywalkerJediKnight.getHp()).toBe(7);
+            });
+
+            it('should give a unit Sentinel for the phase', function () {
```
Comment (AMMayberry1) [inline test/server/cards/03_TWI/events/GuardingTheWay.spec.ts:35]: nit, clarity: should include "when the controller does not have initiative"

File: server/game/cards/03_TWI/events/GuardingTheWay.ts (hunk 1 above)
Comment (supercodepoet) [reply server/game/cards/03_TWI/events/GuardingTheWay.ts:15]: Would it be better to just model the card language? 

`Give Sentinel for the phase. If you have the initiative, also give that unit +2/+2 for this phase.`

File: test/server/cards/03_TWI/events/GuardingTheWay.spec.ts (hunk 5 above)
Comment (supercodepoet) [reply test/server/cards/03_TWI/events/GuardingTheWay.spec.ts:4]: does this work?

`should give a unit Sentinel for the phase and when the controller has initiative also give it +2/+2`

File: test/server/cards/03_TWI/events/GuardingTheWay.spec.ts (hunk 2 above)
Comment (supercodepoet) [reply test/server/cards/03_TWI/events/GuardingTheWay.spec.ts:26]: 👍

File: test/server/cards/03_TWI/events/GuardingTheWay.spec.ts (hunk 4 above)
Comment (supercodepoet) [reply test/server/cards/03_TWI/events/GuardingTheWay.spec.ts:49]: I wanted to simulate play but I can start Chewbacca on the board letting it swing to base then having him swing into Luke when he gains Sentinel.

**Review by AMMayberry1 — APPROVED** [review]

**Pull request merged**